service-csw/src/main/java/org/oskari/csw/request/GetRecords.java

```diff
@@ -0,0 +1,154 @@
+    private final XMLOutputFactory xof = XMLOutputFactory.newFactory();
+    private static final String CSW_URI = "http://www.opengis.net/cat/csw/2.0.2";
+    private static final String GMD_URI = "http://www.isotc211.org/2005/gmd";
+
+    private static final String CSW_VERSION = "2.0.2";
+    private static final String CONSTRAINT_VERSION = "1.1.0";
+
+    /**
+     *
+     <csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" service="CSW" version="2.0.2">
+         <csw:Query typeNames="csw:Record">
+            <ElementSetName typeNames="csw:Record">full</ElementSetName>
+            <csw:Constraint version="1.1.0">
+                ...
+     */
+    public GetRecords() {
+        try {
+            xsw = xof.createXMLStreamWriter(stream);
+            xsw.writeStartDocument();
+            xsw.writeStartElement("csw", "GetRecords", CSW_URI);
+            xsw.writeNamespace("csw", CSW_URI);
+            xsw.writeNamespace("gmd", GMD_URI);
+            xsw.writeNamespace("gml", "http://www.opengis.net/gml");
+            xsw.writeNamespace("ows", "http://www.opengis.net/ows");
+            xsw.writeNamespace("ogc", "http://www.opengis.net/ogc");
+            xsw.writeNamespace("xsi", "http://www.w3.org/2001/XMLSchema-instance");
+            xsw.writeAttribute("service", "CSW");
+            xsw.writeAttribute("version", CSW_VERSION);
+
+            xsw.writeAttribute("maxRecords", "10000");
+            xsw.writeAttribute("startPosition", "1");
+
+            xsw.writeAttribute("resultType", "results"); // or "validate" or "hits"
+            xsw.writeAttribute("outputFormat", "application/xml");
+            xsw.writeAttribute("outputSchema", "http://www.isotc211.org/2005/gmd");
+
+            xsw.writeStartElement(CSW_URI, "Query");
+            //xsw.writeAttribute("typeNames", "csw:Record");
+            xsw.writeAttribute("typeNames", "gmd:MD_Metadata");
+
+            // we need "full" query to get locale mapping like #SW -> swe -> sv
+            // to optimize we could try to do an "init" query to get mappings and use
+            // "summary" query to get the data. Note! Since locale mappings are at "result item" level
+            // this might lead to complications. Just using "full" for now, it's more XML to transfer and
+            // parse but it's safe.
+            xsw.writeStartElement(CSW_URI, "ElementSetName");
+            xsw.writeCharacters("full");
+            xsw.writeEndElement(); // ElementSetName
+
+
+            xsw.writeStartElement(CSW_URI, "Constraint");
+            xsw.writeAttribute("version", CONSTRAINT_VERSION);
+
+        } catch (XMLStreamException e) {
+            throw new ServiceRuntimeException("Couldn't create GetRecords request", e);
+        }
+        filterFactory = CommonFactoryFinder.getFilterFactory();
+    }
+
+    public void setFilters(List<Filter> filters) {
+        this.filters = filters;
+    }
+    public void addEqualFilter(String name, String value) {
```

Comment:
This and the filterFactory will be removed from the class. setFilters is the way to go here since there's some configurable logic how the filters are joined so it's not something that we want in this class.

Reply:
If you're only going to use `setFilters(List<Filter> filters)` then consider `getXML(List<Filter> filters)` (or just `getXML(Filter filter)` with the caller making the decision how multiple Filters should be joined together (now they are ANDed together)

Reply:
True. With the removal of addXX() methods this class can just provide static method to get the XML.